web-platform/website/src/components/pages/account-settings/EditAccountEmailView.tsx

```diff
@@ -0,0 +1,278 @@
+import { FirebaseError } from "@firebase/app";
+import { useEffect, useRef, useState } from "react";
+import {
+  Col,
+  Container,
+  Card,
+  Form,
+  FormFeedback,
+  FormGroup,
+  Input,
+  Label,
+  Row,
+  Button
+} from "reactstrap";
+
+import { style } from "typestyle";
+import {
+  LoginFormValidationResult,
+  validateLoginForm,
+} from "../login/LoginFormValidator";
+import { EmailChangedView } from './EmailChangedView'
+import { Strings } from "../../../resources/Strings";
+import { useAuthentication } from "../../../services/AuthenticationService";
+import { Colors, Spacing } from "../../../styles";
+import { DisabledProductButton, ProductButton, TertiaryButton, TransparentButton } from "../../../styles/Buttons";
+import { ContainerStyles } from "../../../styles/ContainerStyles";
+import { FontSizeRaw, Fonts } from "../../../styles/Fonts";
+import { getFirebaseErrorMessage } from "../../../utils/FirebaseErrors";
+import {
+  AccountSettingsState,
+  useAccountSettingsDataContext,
+} from "./AccountSettingsDataContext";
+
+
+const strings = Strings.components.pages.accountSettings.editEmailAccount;
+const firebaseStings = Strings.utils.firebaseError.errorMessages
+
+export function EditAccountEmailView() {
+  const [confirmationView, setConfirmationView] = useState(false)
+  const [email, setEmail] = useState("");
+  const [isDisabled, setDisabled] = useState(true);
+  const [password, setPassword] = useState("");
+  const [passwordVisible, setPasswordVisbile] = useState(false);
+  const [eyeIconVisible, setVisibility] = useState(false)
+  const [validationResult, setValidationResult] =
+    useState<LoginFormValidationResult>();
+  const { setAccountSettingsState } = useAccountSettingsDataContext();
+
+
+  const isEmailInvalid = Boolean(
+    validationResult && validationResult.email && validationResult.email.error
+  );
+
+  const isPasswordInvalid = Boolean(
+    validationResult &&
+    validationResult.password &&
+    validationResult.password.error
+  );
+
+  // Prevents closure in validateAndUpdate
+  const emailRef = useRef(email);
+  emailRef.current = email;
+
+  const passwordRef = useRef(password);
+  passwordRef.current = password;
+
+  useEffect(() => {
+    setValidationResult((validationResult) => ({
+      ...(validationResult || {
+        email: { error: undefined },
+        password: { error: undefined },
+      }),
+      valid: true,
+      passwordRules: [],
+    }));
+  }, []);
+
+  const { changeUserEmail } = useAuthentication();
+
+  const handleChange = (e: any) => {
+    setValidationResult({
+      email: {},
+      password: {},
+      passwordRules: [],
+      valid: true,
+    });
+
+    setDisabled(false)
+    if (e.target.value === "") {
+      setDisabled(true)
+    }
+    if (e.target.name === "password") {
```

Comment:
Instead of handling all changes in one callback we should have smaller, separate and discrete handlers.